sdks/java/testing/load-tests/src/main/java/org/apache/beam/sdk/loadtests/SideInputLoadTest.java

```diff
@@ -0,0 +1,243 @@
+      sideInput =
+          input
+              .apply(Window.into(FixedWindows.of(Duration.millis(windowDurationMilis))))
+              .apply(View.asIterable());
+    } else {
+      sideInput = input.apply(View.asIterable());
+    }
+
+    input
+        .apply(ParDo.of(new SideInputTestWithIterable(sideInput)).withSideInputs(sideInput))
+        .apply("Collect end time metrics", ParDo.of(runtimeMonitor));
+  }
+
+  private static class AddTimestamps extends DoFn<KV<byte[], byte[]>, KV<byte[], byte[]>> {
+    private static Instant timestamp = TIME;
+
+    @ProcessElement
+    public void processElement(ProcessContext c) {
+      timestamp = TIME.plus(1L);
+      c.outputWithTimestamp(c.element(), timestamp);
+    }
+  }
+
+  private static class SideInputTestWithList extends DoFn<KV<byte[], byte[]>, KV<byte[], byte[]>> {
+
+    private final PCollectionView<List<KV<byte[], byte[]>>> sideInput;
+
+    public SideInputTestWithList(PCollectionView<List<KV<byte[], byte[]>>> sideInput) {
+      this.sideInput = sideInput;
+    }
+
+    @ProcessElement
+    public void processElement(ProcessContext c) {
+      List<KV<byte[], byte[]>> si = c.sideInput(sideInput);
+
+      for (KV<byte[], byte[]> sideInputElement : si) {
+        // for every _input_ element iterate over all _sideInput_ elements
+        // count consumed bytes, examine memory usage, etc (Metrics API).
+        byte[] key = sideInputElement.getKey();
+      }
+    }
+  }
+
+  private class SideInputTestWithHashMap extends DoFn<KV<byte[], byte[]>, KV<byte[], byte[]>> {
+
+    private final PCollectionView<Map<byte[], byte[]>> sideInput;
+
+    public SideInputTestWithHashMap(PCollectionView<Map<byte[], byte[]>> sideInput) {
+      this.sideInput = sideInput;
+    }
+
+    @ProcessElement
+    public void processElement(ProcessContext c) {
+      Map<byte[], byte[]> si = c.sideInput(sideInput);
+      // access Map elements according to options - % of random access
+      ArrayList<byte[]> keyList = new ArrayList<>(si.keySet());
+      int size = keyList.size();
+      int elementCount = size / options.getAccessPercentage() * 100;
+      Random gen = new Random();
+      for (int i = 0; i < elementCount; i++) {
+        byte[] value = si.get(keyList.get(gen.nextInt(elementCount)));
+      }
+    }
+  }
+
+  private class SideInputTestWithIterable extends DoFn<KV<byte[], byte[]>, KV<byte[], byte[]>> {
+
+    private final PCollectionView<Iterable<KV<byte[], byte[]>>> sideInput;
+
+    public SideInputTestWithIterable(PCollectionView<Iterable<KV<byte[], byte[]>>> sideInput) {
+      this.sideInput = sideInput;
+    }
+
+    @ProcessElement
+    public void processElement(ProcessContext c) {
+      Iterable<KV<byte[], byte[]>> si = c.sideInput(sideInput);
+      Integer accessPercentage = options.getAccessPercentage();
+      int elementCount = (int) (sourceOptions.numRecords * accessPercentage / 100);
+      for (KV<byte[], byte[]> sideInputElement : si) {
```

Comment:
For `View.asIterable` it makes the most sense to scan a random distance into the iterable. The most important use case is when you cannot read it all into a list, I think. Both cases are interesting: reading the whole iterable and reading just a prefix.